source/backend/database/models/completedRecipesModel.js

```diff
@@ -13,4 +13,31 @@ async function getCompletedRecipes(userId){
   return completedRecipes;
 }
 
-module.exports = { getCompletedRecipes };
\ No newline at end of file
+/**
+ *  Adds a new recipe to the user's completed list by recipeID
+ * @param recipeId
+ * @returns {Promise<awaited Knex.QueryBuilder<TRecord, TResult>>}
+ */
+ async function addNewRecipe(recipeId, userId){
+  
+  await db.transaction(async (transaction) => {
+    try {
+      await db('completedRecipes')
+            .transacting(transaction)
+            .where({userId, recipeId})
+            .insert();
+      await transaction.commit();
+    } catch (err) {
+      console.log(err);
+      await transaction.rollback();
+    }
+  });
+  
+  const updatedRecipeList = await db('completedRecipes')

```

Comment:
We can delete line 36 - 40 as the usage of the function simply inserts the completeRecipe record, no need to return an updated recipe list

Reply:
I am a little bit confused here. If I were to delete those lines, then how would I be able to insert the recipe into the completeRecipe record?

Reply:
Line 36-40 doesn't add to the completedRecipes table properly because it's query selecting everything from recipes and seems to be filtering the userId wrong? Also, I think you should rename the function from insert -> addToCompletedList or something like that. I think insert can be kinda dangerous to use since the knex uses that function; although it should be fine how you have it. All you should need to do is something like this: 
```
async function addToUsersComplete(recipeId, userId) {
  return await db('completedRecipes')
          .insert({recipeId, userId});
}
```
Try something like that out in the models. I think it's actually pretty identical to what you have already, so just return the updatedRecipeList

Reply:
Actually looking at your code, i don't think you need to return it? but overall I think your adjustments are pretty good. have you tried testing it?

Reply:
Usually there is no need to assign the function returned result to a variable if you are doing insert/update/deletion of a record bc there is nothing to return. so a simple
```
await db('completedRecipes')
          .insert({recipeId, userId});
```
will do. 

If you want to return some insert status to the route to make sure nothing bad happened, you can make a return_status variable, and wrap the above code in a try catch block. If everything successfully executed without error, you set return status to true, if an error happened, then in catch block, set return status to false, and return it at the every end.

Reply:
Thanks for the clarification and explanation guys!